src/game/client/maputil.cpp

```diff
@@ -67,11 +94,848 @@ Utf8String MapCache::Get_User_Map_Dir()
+            throw CODE_05;
+        }
+
+        s_mapDX = s_width - 2 * s_borderSize;
+        s_mapDY = s_height - 2 * s_borderSize;
+        return true;
+    } else {
+        return false;
+    }
+}
+
+unsigned int Calc_CRC(Utf8String dir, Utf8String name)
+{
+    CRC crc;
+    crc.Clear();
+
+    Utf8String str;
+    char buf[PATH_MAX];
+    strcpy(buf, name.Str());
+    int len = strlen(buf);
+    char buf2[PATH_MAX];
+
+    if (len >= 4) {
+        memset(buf2, 0, PATH_MAX);
+        strncpy(buf2, buf, len - 4);
+    }
+
+    str = name;
+    File *file = g_theFileSystem->Open_File(str.Str(), File::READ);
+
+    if (file != nullptr) {
+        for (;;) {
+            unsigned char buf3[4096];
+            int val = file->Read(buf3, 4096);
+
+            if (val <= 0) {
+                break;
+            }
+
+            crc.Compute_CRC(buf3, val);
+        }
+
+        file->Close();
+        return crc.Get_CRC();
+    } else {
+        captainslog_dbgassert(false, "Couldn't open '%s'", name.Str());
+        return 0;
+    }
+}
+
+void Reset_Map()
+{
+    if (s_waypoints != nullptr) {
+        delete s_waypoints;
+        s_waypoints = nullptr;
+    }
+
+    s_techPositions.clear();
+    s_supplyPositions.clear();
+}
+
+void Get_Extent(Region3D *region)
+{
+    region->lo.x = 0.0f;
+    region->lo.y = 0.0f;
+    region->hi.x = s_mapDX * 10.0f;
+    region->hi.y = s_mapDY * 10.0f;
+    region->lo.z = 0.0f;
+    region->hi.z = 0.0f;
+}
+
+bool Would_Map_Transfer(const Utf8String &path)
+{
+    Utf8String str = g_theMapCache->Get_User_Map_Dir();
+    return path.Starts_With_No_Case(str.Str());
+}
+
+int Populate_Map_List_Box_No_Reset(GameWindow *listbox, bool is_official, bool is_multiplayer, Utf8String map_name)
+{
+    if (g_theMapCache != nullptr && listbox != nullptr) {
+        int columns = Gadget_List_Box_Get_Num_Columns(listbox);
+        const Image *star_bronze = nullptr;
+        const Image *star_silver = nullptr;
+        const Image *star_gold = nullptr;
+        const Image *red_yel_star = nullptr;
+        SkirmishBattleHonors *honors = nullptr;
+        int width = 10;
+        int height = 10;
+
+        if (columns > 1) {
+            star_bronze = g_theMappedImageCollection->Find_Image_By_Name("Star-Bronze");
+            star_silver = g_theMappedImageCollection->Find_Image_By_Name("Star-Silver");
+            star_gold = g_theMappedImageCollection->Find_Image_By_Name("Star-Gold");
+            red_yel_star = g_theMappedImageCollection->Find_Image_By_Name("RedYell_Star");
+            honors = new SkirmishBattleHonors();
+            int image_width;
+
+            if (star_gold != nullptr) {
+                image_width = star_gold->Get_Image_Width();
+            } else {
+                image_width = 10;
+            }
+
+            width = image_width;
+            int column_width = Gadget_List_Box_Get_Column_Width(listbox, 0);
+            width = GameMath::Min(column_width, image_width);
```

Comment:
Something not right here. This overrides assignment of 2 lines above.

Reply:
Original does it this way.